src/tree/updater_gpu_hist.cu

```diff
@@ -238,6 +243,43 @@ struct GPUHistMakerDevice {
     }
   }
 
+  void InitFeatureGroups() {
```

Comment:
I'm trying to think of ways to get rid of initialisation and make the kernel self-contained. Is it possible for a thread block to do a binary search over bins/features inside the kernel to independently allocate work to itself? Bin/feature information is available inside EllPackDeviceAccessor.

Reply:
I've moved feature groups into separate data types, this also made writing tests easier.